cfme/tests/infrastructure/test_vm_rest.py

```diff
@@ -6,35 +6,60 @@
 from cfme.rest.gen_data import a_provider as _a_provider
 from cfme.rest.gen_data import vm as _vm
 from cfme.utils import error
-from cfme.utils.rest import assert_response, delete_resources_from_collection
+from cfme.utils.blockers import BZ
+from cfme.utils.rest import (
+    assert_response,
+    delete_resources_from_collection,
+    query_resource_attributes,
+)
 from cfme.utils.wait import wait_for
 
 
 pytestmark = [test_requirements.provision]
 
 
-@pytest.fixture(scope="function")
+@pytest.fixture(scope='function')
 def a_provider(request):
     return _a_provider(request)
 
 
-@pytest.fixture(scope="function")
-def vm_name(request, a_provider, appliance):
-    return _vm(request, a_provider, appliance.rest_api)
+@pytest.fixture(scope='function')
+def vm(request, a_provider, appliance):
+    vm_name = _vm(request, a_provider, appliance.rest_api)
+    return appliance.rest_api.collections.vms.get(name=vm_name)
 
 
+@pytest.mark.tier(3)
+def test_query_vm_attributes(vm, soft_assert):
+    """Tests access to VM attributes using /api/vms.
+
+    Metadata:
+        test_flag: rest
+    """
+    outcome = query_resource_attributes(vm)
+    for failure in outcome.failed:
+        if failure.type == 'attribute' and failure.name == 'policy_events' and BZ(
+                1546995, forced_streams=['5.8', '5.9', 'upstream']).blocks:
+            continue
+        # this one is expected because additional arguments are needed
```

Comment:
**Question**

Is there a test case that includes the additional args to prevent this subcollection failure?

Reply:
@mshriver Not yet, I just added it my TODO list.